Update ssh for more hardening

jpluimers opened the pull request:
I've applied some more `sshd_config` hardening from these sources:

- http://people.redhat.com/swells/mea/SECSCAN-FirstRun/sshd_config.htm
- http://wp.kjro.se/2013/09/06/hardening-your-ssh-server-opensshd_config
- http://kacper.blog.redpill-linpro.com/archives/702
- https://www.freebsd.org/cgi/man.cgi?sshd_config(5)

One of the things it does is remove any leading comment `#` on those lines (I needed this because SuSE has most settings behind comment hashes).

jpluimers committed:
Update ssh for more hardening

I've applied some more `sshd_config` hardening from these sources:

- http://people.redhat.com/swells/mea/SECSCAN-FirstRun/sshd_config.htm
- http://wp.kjro.se/2013/09/06/hardening-your-ssh-server-opensshd_config
- http://kacper.blog.redpill-linpro.com/archives/702
- https://www.freebsd.org/cgi/man.cgi?sshd_config(5)

One of the things it does is remove any leading comment `#` on those lines (I needed this because SuSE has most settings behind comment hashes).

comotion commented on modules/ssh:
```diff
@@ -24,12 +24,27 @@ cd /etc/ssh
+-e 's/#\?ChallengeResponseAuthentication *\(yes\|no\).*/ChallengeResponseAuthentication yes/' \
+-e 's/#\?KerberosAuthentication *\(yes\|no\).*/KerberosAuthentication no/' \
+-e 's/#\?GSSAPIAuthentication *\(yes\|no\).*/GSSAPIAuthentication no/' \
+-e 's/#\?GatewayPorts *\(yes\|no\).*/GatewayPorts no/' \
```
Most of the improvements are sane, but sometimes X11 forwarding and gateway ports are used, so setting this as a global policy will get in the way.

jpluimers replied:
So if I amend this merge request with a commit that:

1. comments out the lines with `GatewayPorts ` and `X11Forwarding `, 
2. adding a descriptive comment text why, 

the merge request is OK for you?

comotion replied:
yes, I like the rest.

jpluimers replied:
Done. See the new commit.

jpluimers committed:
No auto-disable of GatewayPorts and X11Forwarding

Do not automatically disable `GatewayPorts` and `X11Forwarding` as sometimes they are used.

Added them as a commented `sed` call so users can uncomment when they don't use `GatewayPorts` and `X11Forwarding`.

jpluimers committed:
Left too much of the review comment

Forgot to remove the review comment. Now it's gone.
